op_crates/webgpu/binding.rs

```diff
@@ -9,23 +9,54 @@ use deno_core::ZeroCopyBuf;
 use deno_core::{OpState, Resource};
 use serde::Deserialize;
 use std::borrow::Cow;
+use std::rc::Rc;
+
+use crate::Instance;
 
 use super::error::WebGpuError;
 
-pub(crate) struct WebGpuBindGroupLayout(
-  pub(crate) wgpu_core::id::BindGroupLayoutId,
-);
+pub(crate) struct WebGpuBindGroupLayout {
+  pub instance: Rc<Instance>,
+  pub device: Rc<wgpu_core::id::DeviceId>,
+  pub bind_group_layout: Rc<wgpu_core::id::BindGroupLayoutId>,
+}
 impl Resource for WebGpuBindGroupLayout {
   fn name(&self) -> Cow<str> {
     "webGPUBindGroupLayout".into()
   }
+
+  fn close(self: Rc<Self>) {
+    let resource = Rc::try_unwrap(self)
+      .map_err(|_| "closed webGPUBindGroupLayout while in use")
+      .unwrap();
```

Comment:
It is obviously not great that we panic here, but a) we can't return errors from `Resource#close` calls, and b) these resources should never be closed by the user directly. If they do and Deno panics, they might learn to not do that :-)